### DISCUSSION TITLE
PDF++ as default viewer for PC

### DISCUSSION THREAD

**Discussion by LeonelRFF** [RyotaUshio/obsidian-pdf-plus#271] [Q&A]
Hi, I've seen the wiki and I see that there is the possibility of opening the PDFs that are seen on the web in obsidian https://ryotaushio.github.io/obsidian-pdf-plus/bookmarklet-to-create-dummy-pdf-files.html , I managed to do that, my question is regarding what is mentioned in https://ryotaushio.github.io/obsidian-pdf-plus/external-pdf-files.html Is it possible that from the operating system file explorer when opening a PDF it opens directly in PDF++ within obsidian? or is it something that should be requested directly to obsidian so that this can be done in some way, thanks in advance

**Comment by N3C2L** [comment]
Are you familiar with macro software like autohotkey or do you have a keyboard software installed where you can create macros? Because in my case I have an autohotkey macro which makes this actions in one key: Assuming you have the .pdf file selected in the file explorer: F2 for renaming the title > Copy > Switch to Obsidian > Paste to quick switcher ++ addon for a .pdf search > Enter.

**Reply by LeonelRFF** [reply]
Hi, yes, I know and use Autohotkeys, I thought about that alternative, although my idea is more to open the PDF by double clicking or pressing enter, but I couldn't do this.

**Comment by RyotaUshio** [comment]
I don't think you can set Obsidian as the system-wide default PDF viewer, because in order to open a file in Obsidian, you first need to create a vault that contains that file.

So the best you could do would be some macro-ish stuff. I'm not familiar with AutoHotkey and the like, but you might be able to use Obsidian URI, specifically the `path` parameter.
https://help.obsidian.md/Extending+Obsidian/Obsidian+URI

Also, you can use the `obsidian://pdf-plus?create-dummy=file://<PATH>` uri to create a dummy file for any local PDF file, although I think it is better to create a symbolic link instead.

**Reply by LeonelRFF** [reply]
Thanks for replying, I did not take that information into account, I will try it this way, at least for now the PDFs I see are in my vault.
